nemo/src/builder_proxy.rs

```diff
@@ -11,286 +18,723 @@ use nemo_physical::{
     datatypes::Double,
 };
 
-use oxiri::Iri;
-use rio_api::parser::TriplesParser;
-use rio_turtle::TurtleParser;
-
 use crate::{
     error::ReadingError,
-    io::{
-        formats::rdf_triples::TurtleEncodedRDFTerm,
-        parser::{parse_bare_name, span_from_str},
-    },
+    io::parser::{parse_bare_name, span_from_str},
 };
 
-/// Trait capturing builder proxies that use plain string (used for parsing in logical layer)
-///
-/// This parses from a given logical type to the physical type, without exposing details from one layer to the other.
-pub trait LogicalColumnBuilderProxy<'a, 'b>: ColumnBuilderProxy<String> {
-    /// Create a new [`LogicalColumnBuilderProxy`] from a given [`BuilderProxy`][nemo_physical::builder_proxy::PhysicalBuilderProxyEnum].
-    ///
-    /// # Panics
-    /// If the logical and the nested physical type are not compatible, an `unreachable` panic will be thrown.
-    fn new(physical_builder_proxy: &'b mut PhysicalBuilderProxyEnum<'a>) -> Self
-    where
-        Self: Sized;
-}
+use super::model::types::primitive_logical_value::{
+    any_double_to_physical_double, any_integer_to_physical_integer, any_string_to_physical_string,
+    any_term_to_physical_string, logical_double_to_physical_string,
+    logical_integer_to_physical_string, logical_string_to_physical_string,
```

Comment:
I wonder whether a newtype `PhysicalString(…)` and `From`/`TryFrom` impls would be easier to read?

Reply:
Done!
This got a little out of hand though introducing wrapper structs for many logical types as well.
(Note that I cannot implement e.g. `From<i64> for PhysicalString` since both `i64` and `PhysicalString` are not in the logical crate.)